allennlp/training/metrics.py

```diff
@@ -0,0 +1,190 @@
+from typing import Optional, Tuple, Union
+
+from overrides import overrides
+import torch
+
+from allennlp.common.registrable import Registrable
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+
+
+class Metric(Registrable):
+    """
+    A very general abstract class representing a metric which can be
+    accumulated.
+    """
+    def __call__(self, predictions: torch.Tensor,
+                 gold_labels: torch.Tensor,
+                 mask: Optional[torch.Tensor]):
```

Comment:
Brief docstring would be nice, where the point is to tell someone who's implementing a `Metric` what they should expect to receive when implementing this method.